mart/attack/callbacks/base.py

```diff
@@ -6,27 +6,74 @@
 #
 
 import abc
+from typing import Any, Dict, Union
+
+import torch
+
+from mart.nn import SequentialDict
 
 __all__ = ["Callback"]
 
 
 class Callback(abc.ABC):
     """Abstract base class of callbacks."""
 
-    def on_run_start(self, adversary, input, target, model, **kwargs):
+    def on_run_start(
+        self,
+        adversary: torch.nn.Module,
```

Comment:
This should be `Adversary`.

Reply:
For the moment I will leave it as `torch.nn.Module` due to a circular import issue when tryin to use the `Adversary` type in this file.